docs/writing-rules/pattern-syntax.md

```diff
@@ -421,6 +424,20 @@ It will not match the following:
 import foo
 ```
 
+### Statements as expressions
+
+`foo()` is an expression
+
+`foo();` is a statement.
+
+If youre search pattern is a statment, Semgrep will automatically try to search for it as _both_ an expression and a statement. 
+
+When you write the expression `foo()` in a pattern, Semgrep will visit every expression and sub-expression in your program and try to find a match. 
+
+What about a statement like `foo();`? (Note that `foo()` is confusingly called an expression statement, but it is first a statement). Many programmers don't really see the difference between `foo()` and `foo();`. This is why when one looks for `foo()`; Semgrep thinks the user wants to match statements like `a = foo();`, or `print(foo());`.
```

Comment:
>What about a statement like `foo();`? (Note that `foo()` is confusingly called an expression statement, but it is first a statement). Many programmers don't really see the difference between `foo()` and `foo();`. This is why when one looks for `foo()`; Semgrep thinks the user wants to match statements like `a = foo();`, or `print(foo());`.

Is the semicolon in the last sentence (just after `foo()`) needed? I think it's throwing me off here.

Reply:
In some languages such as C yes, you need the semicolon. In python no, but you need a newline, which is treated internally like a semicolon, as a way to terminate a statement.